enterprise/server/workflow/service/service.go

```diff
@@ -1575,11 +1574,28 @@ func (ws *workflowService) createQueuedStatus(ctx context.Context, wf *tables.Wo
 	}
 	invocationURL += "?queued=true"
 	status := github.NewGithubStatusPayload(actionName, invocationURL, "Queued...", github.PendingState)
-	provider, err := ws.providerForRepo(wd.TargetRepoURL)
+	baseURL := getBaseURL(wd)
+	provider, err := ws.providerForRepo(baseURL)
 	if err != nil {
 		return err
 	}
-	return provider.CreateStatus(ctx, wf.AccessToken, wd.TargetRepoURL, wd.SHA, status)
+	return provider.CreateStatus(ctx, wf.AccessToken, baseURL, wd.SHA, status)
+}
+
+// getBaseURL returns the primary URL the workflow should be attributed to.
+func getBaseURL(wd *interfaces.WebhookData) string {
```

Comment:
nit: should we call this 'statusReportingURL' ? I could see 'baseURL' getting confused with TargetRepoURL